ci-operator/step-registry/fips-check/fips-or-die/fips-check-fips-or-die-commands.sh

```diff
@@ -111,6 +111,22 @@ images=("quay.io/openshift-qe-optional-operators/myapp:v1.16-1"
 set_proxy
 run_command "oc whoami"
 run_command "oc version -o yaml"
+
+version=`oc get clusterversion version -o=jsonpath="{.status.desired.version}"`
+if [[ $version =~ "multi" ]]; then
+    echo "This is a multi-arch cluster, skip it."
+    exit 0
+fi
+
+run_command "oc extract secret/pull-secret -n openshift-config --confirm --to /tmp"; ret=$?
+if [[ $ret -eq 0 ]]; then
+    auths=`cat /tmp/.dockerconfigjson`
+    if [[ $auths =~ "5000" ]]; then
+        echo "This is a disconnected env, skip it."
+        exit 0
+    fi
+fi
+
```

Comment:
Remove the skip below on the CP's architecture in favour of the one above.

We should skip only if we find an arm64 node (either worker or master).

Starting at line 130

Reply:
it shold be both . if we are planning to support P/Z in the future  the condition here should be cp arch != worker arch then skip

Reply:
Yes, let me rephrase. Rather than executing two checks (1 for single-arch non-supported arches, another for multi), we can execute just one like in:
```
nodes=get_all_nodes()
for n in nodes:
  if  "$NON_SUPPORTED_ARCHES" == *n['metadata']['labels']['kubernetes.io/arch']*:
     echo "found node with arch {n['metadata']['labels']['kubernetes.io/arch']}. Skipping test"
     exit 0

```

this will be valid for both single-arch and multi clusters